### ISSUE TITLE
Mocks generated for Lazy loaded Jpa repository calls don't serialize [Failed to serialize object -> failed to lazily initialize a collection of role]

### ISSUE THREAD

**Issue by TheAmg** [unloggedio/unlogged-sdk#97]
### Describe the bug

On checking the serialized value for mocks generated a "failed to serialize trace ..." will be it's value instead of the expected value. The service level mocks are generated correctly. Mock on repo level is the only thing that breaks.

Can be recreated on -

- Project : `https://github.com/dguduguntla/spring-jpa-lazyload`
- Class : `DepartmentServiceImpl`
- Method : `getAllDepartments`

Before Direct Invoking this method, make sure that `@Transactional` is removed from `DepartmentServiceImpl`.

<img width="1440" alt="Screenshot 2024-07-19 at 10 41 14 AM" src="https://github.com/user-attachments/assets/ee1f8abb-d50e-4f0b-b246-edb7858bf686">

### Reproduction steps

1. Remove `@Transactional` from above mentioned class before starting the project.
2. DirectInvoke the method mentioned above.
3. Check the value of the mock generated for the repo call. [save to get trace]

(2nd start)
https://drive.google.com/file/d/1NG9jq7xuNfp9iAayoW1V1Q2ApiZdCjU5/view?usp=sharing

### Expected behavior

For the mock generated in the repo level to be correct.

### Additional context

Saved json, with trace :
```
{
  "classname" : "com.gdp.lazyload.repository.DepartmentRepository",
  "storedCandidateMap" : { },
  "declaredMockMap" : {
    "com.gdp.lazyload.repository.DepartmentRepository#findAll#()Ljava/util/List<Lcom/gdp/lazyload/entity/Department;>;" : [ {
      "id" : "0739350b-f134-46b0-9dd4-c4b9ca878700",
      "name" : "mock response for call to deptRepo.findAll()",
      "fieldTypeName" : "com.gdp.lazyload.repository.DepartmentRepository",
      "sourceClassName" : "com.gdp.lazyload.service.DepartmentServiceImpl",
      "fieldName" : "deptRepo",
      "methodName" : "findAll",
      "whenParameter" : [ ],
      "thenParameter" : [ {
        "returnParameter" : {
          "declaredMocks" : [ ],
          "value" : "\"{\\\"error\\\": \\\"failed to serialize object\\\", \\\"message\\\":\\\"failed to lazily initialize a collection of role: com.gdp.lazyload.entity.Department.employees, could not initialize proxy - no Session (through reference chain: java.util.ArrayList[0]->com.gdp.lazyload.entity.Department[\\\"employees\\\"])\\\"}\"",
          "returnValueType" : "REAL",
          "className" : "java.util.List<com.gdp.lazyload.entity.Department>"
        },
        "methodExitType" : "NORMAL"
      } ],
      "methodHashKey" : "com.gdp.lazyload.repository.DepartmentRepository#findAll#()Ljava/util/List<Lcom/gdp/lazyload/entity/Department;>;"
    } ]
  }
}
```

**Comment by kartikeytewari-ul** [comment]
This is resolved when annotation `@Transactional` is added to the method or class under consideration.

**Comment by kartikeytewari-ul** [comment]
This is resolved with SDK v0.6.9. Now direct invoke and mocks will work without adding a Transactional annotation to the code also. A [demo](https://drive.google.com/file/d/19j_Jz5dzwM11UwsMksqQTvYPx8hkU_kL/view?usp=sharing) is here.
